cirq-core/cirq/sim/simulator_base_test.py

```diff
@@ -77,12 +88,15 @@ class CountingSimulator(
         CountingStepResult, CountingTrialResult, CountingActOnArgs, CountingActOnArgs
     ]
 ):
-    def _create_act_on_args(
+    def _create_act_on_arg(
```

Comment:
Please add tests for the new `SimulatorBase`-level behaviors, including:
* Merging states on 2-qubit gates
* Extracting states on measurements
* The various `_create_act_on_args` behaviors: integer/array `initial_state`, true/false `_split_untangled_states`